### ISSUE TITLE
disabling quality/reflections/texture_array_reflections causes OpenGL error ID=1282 to be emitted [GLES3]

### ISSUE THREAD

**Issue by ginobean** [godotengine/godot#40444]
Godot version 3.2.2.stable

Linux
Description:	Ubuntu 20.04 LTS
Release:	20.04
Codename:	focal

using GLES3 renderer.
**OS/device including version:**

Distributor ID:	UbuntuOpenGL ES 3.0 Renderer: GeForce GT 710/PCIe/SSE2

**Issue description:**

First, enable verbose logging, by setting project setting: 
  settings/stdout/verbose_stdout=true

Then, I set the project setting:
   quality/reflections/texture_array_reflections=false

This setting seems to cause the following error message to be emitted, when the project is then loaded or run:

    ERROR: _gl_debug_print: GL ERROR: Source: OpenGL	Type: Error	ID: 1282Severity: High	Message: GL_INVALID_OPERATION error generated. <texture> does not name a layered texture.

What I had expected was that no such error would be generated..

**Steps to reproduce:**
I was able to reproduce this same issue using this sample project: http://kidscancode.org/godot_recipes/files/2d_touch_camera.zip 

**Minimal reproduction project:**

I've uploaded this same zip project to this bug report, for convenience..

[2d_touch_camera.zip](https://github.com/godotengine/godot/files/4931369/2d_touch_camera.zip)

**Comment by clayjohn** [comment]
I can not reproduce. Using Windows 10, RX 5600 XT.

This may be GPU/Driver related. So please ensure you have the latest drivers for your system.

Also note for future bug reports. An MRP is the smallest possible project that can be used to reproduce a bug. The project you included was a completed tutorial project and it didnt even have the settings that you said are required to reproduce the bug.

**Comment by akien-mga** [comment]
I can reproduce the issue in 3.2 (89f57ae12244f3269c9e3fe4684e16ec1fd2c989) with the provided project (after enabling verbose stdout and disabling texture array reflections):
```
ERROR: _gl_debug_print: GL ERROR: Source: OpenGL        Type: Error     ID: 1   Severity: High  Message: GL_INVALID_OPERATION in glFramebufferTextureLayer(invalid texture target GL_TEXTURE_2D)
   At: drivers/gles3/rasterizer_gles3.cpp:123.
ERROR: _gl_debug_print: GL ERROR: Source: OpenGL        Type: Error     ID: 1   Severity: High  Message: GL_INVALID_OPERATION in glFramebufferTextureLayer(invalid texture target GL_TEXTURE_2D)
   At: drivers/gles3/rasterizer_gles3.cpp:123.
ERROR: _gl_debug_print: GL ERROR: Source: OpenGL        Type: Error     ID: 1   Severity: High  Message: GL_INVALID_OPERATION in glFramebufferTextureLayer(invalid texture target GL_TEXTURE_2D)
   At: drivers/gles3/rasterizer_gles3.cpp:123.
ERROR: _gl_debug_print: GL ERROR: Source: OpenGL        Type: Error     ID: 1   Severity: High  Message: GL_INVALID_OPERATION in glFramebufferTextureLayer(invalid texture target GL_TEXTURE_2D)
   At: drivers/gles3/rasterizer_gles3.cpp:123.
ERROR: _gl_debug_print: GL ERROR: Source: OpenGL        Type: Error     ID: 1   Severity: High  Message: GL_INVALID_OPERATION in glFramebufferTextureLayer(invalid texture target GL_TEXTURE_2D)
   At: drivers/gles3/rasterizer_gles3.cpp:123.
ERROR: _gl_debug_print: GL ERROR: Source: OpenGL        Type: Error     ID: 1   Severity: High  Message: GL_INVALID_OPERATION in glFramebufferTextureLayer(invalid texture target GL_TEXTURE_2D)
   At: drivers/gles3/rasterizer_gles3.cpp:123.
```
The "ID: " part changes at each run.

This is on Mageia 8 with `AMD VEGAM (DRM 3.37.0, 5.7.9-desktop-1.mga8, LLVM 10.0.0)`.
Same issue with `Mesa Intel(R) HD Graphics 630 (KBL GT2)` too.

**Comment by akien-mga** [comment]
It's reproducible in a new project with the `default_env.tres`. Disabling the default env removes the warning, so it's triggered by one of the features of the environment.

**Comment by akien-mga** [comment]
Backtrace for the error:
```
(gdb) bt
#0  _gl_debug_print (source=33350, type=33356, id=1, severity=37190, length=87, message=0x7fffffffb680 "GL_INVALID_OPERATION in glFramebufferTextureLayer(invalid texture target GL_TEXTURE_2D)", userParam=0x0)
    at drivers/gles3/rasterizer_gles3.cpp:123
#1  0x00007ffff4c54d4a in _mesa_error () from /usr/lib64/dri/radeonsi_dri.so
#2  0x00007ffff4c595a9 in check_texture_target () from /usr/lib64/dri/radeonsi_dri.so
#3  0x00007ffff4c5f8a9 in _mesa_FramebufferTextureLayer () from /usr/lib64/dri/radeonsi_dri.so
#4  0x0000000001fa1384 in RasterizerStorageGLES3::sky_set_texture (this=0x57b52b0, p_sky=..., p_panorama=..., p_radiance_size=128) at drivers/gles3/rasterizer_storage_gles3.cpp:2172
#5  0x000000000368353e in VisualServerRaster::sky_set_texture (this=0x57cce90, arg1=..., arg2=..., arg3=128) at servers/visual/visual_server_raster.h:179
#6  0x00000000036b7bd9 in VisualServerWrapMT::sky_set_texture (this=0x56180f0, p1=..., p2=..., p3=128) at servers/visual/visual_server_wrap_mt.h:115
#7  0x0000000003335b63 in ProceduralSky::_radiance_changed (this=0x594d8a0) at scene/resources/sky.cpp:130
#8  0x0000000003336d8b in ProceduralSky::_update_sky (this=0x594d8a0) at scene/resources/sky.cpp:420
#9  0x00000000014cfce8 in MethodBind0::call (this=0x651a7a0, p_object=0x594d8a0, p_args=0x0, p_arg_count=0, r_error=...) at ./core/method_bind.gen.inc:59
#10 0x00000000038e39ca in Object::call (this=0x594d8a0, p_method=..., p_args=0x0, p_argcount=0, r_error=...) at core/object.cpp:922
#11 0x00000000038da61c in MessageQueue::_call_function (this=0x55ed0c0, p_target=0x594d8a0, p_func=..., p_args=0x7ffff625b038, p_argcount=0, p_show_error=false) at core/message_queue.cpp:250
#12 0x00000000038da8e3 in MessageQueue::flush (this=0x55ed0c0) at core/message_queue.cpp:297
#13 0x0000000002b1b56b in SceneTree::iteration (this=0x5907ad0, p_time=0.0166666675) at scene/main/scene_tree.cpp:483
#14 0x00000000014628be in Main::iteration () at main/main.cpp:2084
#15 0x0000000001431432 in OS_X11::run (this=0x7fffffffcf50) at platform/x11/os_x11.cpp:3233
#16 0x00000000014216ac in main (argc=1, argv=0x7fffffffd778) at platform/x11/godot_x11.cpp:56
```

**Comment by clayjohn** [comment]
Cheers for strict drivers on Linux. I have a feeling this may solve an issue on Android as well.

**Comment by clayjohn** [comment]
Closed by #40558
